example/synchronous_resnet/resnet_worker.py

```diff
@@ -0,0 +1,566 @@
+    """Creates two-branch residual block
+
+    Parameters:
+    ----------
+    incoming_layer : instance of Lasagne layer
+        Parent layer
+
+    ratio_n_filter : float
+        Scale factor of filter bank at the input of residual block
+
+    ratio_size : float
+        Scale factor of filter size
+
+    has_left_branch : bool
+        if True, then left branch contains simple block
+
+    upscale_factor : float
+        Scale factor of filter bank at the output of residual block
+
+    ix : int
+        Id of residual block
+
+    Returns
+    -------
+    tuple: (net, last_layer_name)
+        net : dict
+            Dictionary with stacked layers
+        last_layer_name : string
+            Last layer name
+    """
+    net = {}
+
+    # right branch
+    net_tmp, last_layer_name = build_simple_block(
+        incoming_layer, map(lambda s: s % (ix, 2, 'a'), simple_block_name_pattern),
+        int(lasagne.layers.get_output_shape(incoming_layer)[1]*ratio_n_filter),
+        1, int(1.0/ratio_size), 0)
+    net.update(net_tmp)
+
+    net_tmp, last_layer_name = build_simple_block(
+        net[last_layer_name], map(lambda s: s % (ix, 2, 'b'), simple_block_name_pattern),
+        lasagne.layers.get_output_shape(net[last_layer_name])[1], 3, 1, 1)
+    net.update(net_tmp)
+
+    net_tmp, last_layer_name = build_simple_block(
+        net[last_layer_name], map(lambda s: s % (ix, 2, 'c'), simple_block_name_pattern),
+        lasagne.layers.get_output_shape(net[last_layer_name])[1]*upscale_factor, 1, 1, 0,
+        nonlin=None)
+    net.update(net_tmp)
+
+    right_tail = net[last_layer_name]
+    left_tail = incoming_layer
+
+    # left branch
+    if has_left_branch:
+        net_tmp, last_layer_name = build_simple_block(
+            incoming_layer, map(lambda s: s % (ix, 1, ''), simple_block_name_pattern),
+            int(lasagne.layers.get_output_shape(incoming_layer)[1]*4*ratio_n_filter),
+            1, int(1.0/ratio_size), 0,
+            nonlin=None)
+        net.update(net_tmp)
+        left_tail = net[last_layer_name]
+
+    net['res%s' % ix] = ElemwiseSumLayer([left_tail, right_tail], coeffs=1)
+    net['res%s_relu' % ix] = NonlinearityLayer(net['res%s' % ix], nonlinearity=rectify)
+
+    return net, 'res%s_relu' % ix
+
+
+def build_resnet():
+    net = {}
+    net['input'] = InputLayer((None, 3, 224, 224))
+    sub_net, parent_layer_name = build_simple_block(
+        net['input'], ['conv1', 'bn_conv1', 'conv1_relu'],
+        64, 7, 2, 3, use_bias=True)
+    net.update(sub_net)
+    net['pool1'] = PoolLayer(net[parent_layer_name], pool_size=3, stride=2,
+                             pad=0, mode='max', ignore_border=False)
+
+    block_size = list('abc')
+    parent_layer_name = 'pool1'
+    for c in block_size:
+        if c == 'a':
+            sub_net, parent_layer_name = build_residual_block(net[parent_layer_name],
+                                                              1, 1, True, 4, ix='2%s' % c)
```

Comment:
I've copied the code for building the network, and when I run it, it breaks like so:

```
----> 1 net = build_resnet()

/home/adam/GitRepos/synkhronos/demos/resnet/build_resnet.py in build_resnet()
    151         if c == 'a':
    152             sub_net, parent_layer_name = build_residual_block(net[parent_layer_name],
--> 153                                                               1, 1, True, 4, ix='2%s' % c)
    154         else:
    155             sub_net, parent_layer_name = build_residual_block(net[parent_layer_name],

/home/adam/GitRepos/synkhronos/demos/resnet/build_resnet.py in build_residual_block(incoming_layer, ratio_n_filter, ratio_size, has_left_branch, upscale_factor, ix)
    103         incoming_layer, map(lambda s: s % (ix, 2, 'a'), simple_block_name_pattern),
    104         int(lasagne.layers.get_output_shape(incoming_layer)[1]*ratio_n_filter),
--> 105         1, int(1.0/ratio_size), 0)
    106     net.update(net_tmp)
    107 

/home/adam/GitRepos/synkhronos/demos/resnet/build_resnet.py in build_simple_block(incoming_layer, names, num_filters, filter_size, stride, pad, use_bias, nonlin)
     49     net = []
     50     net.append((
---> 51             names[0],
     52             ConvLayer(incoming_layer, num_filters, filter_size, stride, pad,
     53                       flip_filters=False, nonlinearity=None) if use_bias

TypeError: 'map' object is not subscriptable

```

Reply:
Do you use Python 3? Probably this code isn't python 3 compatible. See 
https://stackoverflow.com/questions/6800481/python-map-object-is-not-subscriptable
for a way to fix this. Or try with python 2.

Reply:
OK thanks, I wrapped each of the four `map()` like so: `list(map())` and now it seems to work.